project/firstproject/src/main/java/com/example/firstproject/services/AuthenticationServiceImpl.java

```diff
@@ -59,6 +59,36 @@ public void saveUser(User user)
         userRepository.save(user);
     }
 
+    @Override
+    public String loginUser(String email, String password)
+    {
+        StringBuilder sb = new StringBuilder();
+        if(email ==  null || !checkEmail(email))
+        {
+            sb.append("Invalid E-Mail!\n");
+        }
+        if(password == null || !checkPassword(password))
+        {
+            sb.append("Password Validation Error!\n");
+        }
+        if(sb.length() == 0)
+        {
+            User dbUser = userRepository.findByEmail(email);
+            if(dbUser == null)
+            {
+                sb.append("User Not Found!\n");
+            }
+                else
```

Comment:
This should be changed to `else if` block because there is only one check:
` else if(!passwordEncoder.matches(password,dbUser.getPassword()))`